### PR TITLE
fix(strip): Combine typescript_class_properties() into strip()

### PR DESCRIPTION
**Pull request by nayeemrmn**
@kdy1 This implements `use_define_for_class_fields: true` as correctly specified in #1472. It also adds the behaviour of the `typescript_class_properties()` pass directly into `strip()` in case of `use_define_for_class_fields: false`. This aligns with tsc. The `typescript_class_properties()` is redundant and should be removed.

However, I can't figure out how to copy the static class field handling from `typescript_class_fields()` to `strip()`. So I left some TODOs and haven't removed it yet. Help would be appreciated.

### CODE REVIEW

File: ecmascript/transforms/typescript/src/strip.rs [hunk 1]
```diff
@@ -145,141 +153,100 @@ impl Strip {
...
-                        expr: define_property_call,
-                    });
-
-                    define_property_stmts.push(stmt);
+                    assign_exprs.push(assign_expr);
+                }
+                ClassMember::ClassProp(_) => {
+                    // TODO(nayeemrmn): Static fields should also be moved to
+                    // assignments after the class. Figure out how. They are
+                    // preserved for now.
```
Comment (nayeemrmn) [inline ecmascript/transforms/typescript/src/strip.rs:225]: Implementation.

File: ecmascript/transforms/typescript/tests/strip.rs [hunk 2]
```diff
@@ -746,6 +748,8 @@ test!(
 
 test!(
     ::swc_ecma_parser::Syntax::Typescript(Default::default()),
+    // TODO(nayeemrmn): This output should be achieved without
+    // `typescript_class_properties()`.
```
Comment (nayeemrmn) [inline ecmascript/transforms/typescript/tests/strip.rs:752]: Test.

**Review by nayeemrmn — COMMENTED** [review]
TODOs for static class field handling when `use_define_for_class_fields: false`.

**Comment by kdy1** [comment]
Sorry for late response. I was too busy because of other projects.

The changes seems ok to me. But I have a question.
Doesn't this change the default behavior of swc even when target < es2020 ?

**Comment by nayeemrmn** [comment]
> Doesn't this change the default behavior of swc even when target < es2020 ?

Not totally sure what you mean by "when target < es2020",  but any change in behaviour is non-breaking and aligns with tsc. The behaviour of `chain!(strip(), typescript_class_properties())` should be exactly the same.

**Comment by kdy1** [comment]
@nayeemrmn Thank you very much for the pr and clarification :)

**Review by kdy1 — APPROVED** [review]
LGTM.
I'll merge it today.

**Pull request merged**